nameserver/zone.go

```diff
@@ -121,6 +122,37 @@ func (zone *ZoneDb) AddRecord(ident string, name string, ip net.IP) error {
 	return nil
 }
 
+func (zone *ZoneDb) DeleteRecords(ident *string, ip *net.IP, name *string) (count int) {
+	zone.mx.Lock()
+	defer zone.mx.Unlock()
+
+	var fqdn *string
+	if name != nil {
+		temp := dns.Fqdn(*name)
+		fqdn = &temp
+	}
+
+	match := func(r dbRecord) bool {
+		return (ident == nil || r.Ident == *ident) &&
+			(ip == nil || r.IP.Equal(*ip)) &&
+			(fqdn == nil || r.Name == *fqdn)
+	}
+
+	w := 0 // write index
+
+	for _, r := range zone.recs {
+		if !match(r) {
+			zone.recs[w] = r
+			w++
+		}
+	}
+
+	count = len(zone.recs) - w
+	zone.recs = zone.recs[:w]
+
+	return
+}
+
```

Comment:
Maybe these use cases are not easy to implement now, but they can be optimally implemented when #615 is merged, as the underlying data structure would be a tree...

Maybe the `(*, *, *)` case makes sense... Or maybe it should be explicitly done in a `Flush()` operation or something similar...

And, regarding returning error codes, maybe we could `panic()` if you find a *no-wildcard* after processing a *wildcard*...